nntrainer/tensor/manager.cpp

```diff
@@ -601,9 +539,9 @@ std::vector<Tensor *> Manager::requestWeightOptimizerVariables(
   ret.reserve(dims.size());
 
   for (unsigned int idx = 0; idx < dims.size(); idx++)
-    ret.push_back(tensor_pool.requestTensor(dims[idx], exec_order, lifespan,
-                                            name + ":opt" + std::to_string(idx),
-                                            initializer));
+    ret.push_back(tensor_pool.request(name + ":opt" + std::to_string(idx),
```

Comment:
opt is currently requested to the tensor pool.  Either should work the same, I think, though

Reply:
In optimization point, I think if the opt tensor is temporal, then it might be good to be tensor pool, but if it is like weight, then I guess it need to be placed in weight. but WM and WV need to be stores like weight. Aren't they?

Reply:
@kparichay Any idea on this?

Reply:
The reason to have weight_pool separate is to have weights allocate at initialize time. But tensor_pool is allocated at train/inference time. As optimizer weights are not needed at initialize time, I am not sure if its a good idea to merge them to weight_pool.

Independent of the above comment, the weight_pool and tensor_pool will later be merged (with some ammendments to Model::Save() and Model::Load()).